src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -897,14 +867,7 @@ void c_typecheck_baset::typecheck_expr_symbol(exprt &expr)
 void c_typecheck_baset::typecheck_side_effect_statement_expression(
   side_effect_exprt &expr)
```

Comment:
If we had something like a `side_effect_statement_exprt` or the likes this could have a better signature and the code code -> expression -> code type conversion in `to_code(to_unary_expr(expr).op())` wouldn't need to take place.

Reply:
yes, agreed

Reply:
The signature change should be separate.